js/widgets/forms/clearButton.js

```diff
@@ -20,59 +25,56 @@ define( [
 		_create: function() {
 			this._super();
 
+			if ( !this.inputNeedsWrap ) {
+				return;
+			}
```

Comment:
All of this should be short circuited by `if ( !this.options.clearBtn || !this.isSearch ) { return; }`